### PR TITLE
FINERACT-1375:ClientEntityPerson

### PR DESCRIPTION
**Pull request by rrpawar96**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by francisguchie** [comment]
Thanks @rrpawar96  
@awasum  @ptuomola  please note that i have already tested the functionality of this and works for me like

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java [hunk 1]
```diff
@@ -1336,6 +1344,16 @@ private String dataScopedSQL(final String appTable, final Long appTableId) {
...
+        if (appTable.equalsIgnoreCase("m_client_person")) {
+            scopedSQL = "select o.id as officeId, null as groupId, c.id as clientId, null as savingsId, null as loanId, null as entityId from m_client c "
+                    + " join m_office o on o.id = c.office_id and o.hierarchy like '" + currentUser.getOffice().getHierarchy() + "%'"
+                    + " where c.id = " + appTableId;
+        }
+        if (appTable.equalsIgnoreCase("m_client_entity")) {
+            scopedSQL = "select o.id as officeId, null as groupId, c.id as clientId, null as savingsId, null as loanId, null as entityId from m_client c "
+                    + " join m_office o on o.id = c.office_id and o.hierarchy like '" + currentUser.getOffice().getHierarchy() + "%'"
+                    + " where c.id = " + appTableId;
+        }
```
Comment (ptuomola) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: Aren't these two SQL identical? If yes then can you combine them to one if with or condition?

**Comment by rrpawar96** [comment]
@ptuomola Please review this PR, This PR resolves the issue by adding a parameter called legalFormName,  while creating Datatables, this param will store value either PERSON or ENTITY, to show the respective datatables to the respective clients (type: person or entity).

**Comment by ptuomola** [comment]
Hi @rrpawar96 - instead of making this very specific to m_client by calling the parameter legalFormName, should we make this a bit more generic? I.e. if you call the new parameter / column something like 'entitySubType' then it could be reused for other tables as well (e.g. only valid for some offices / loans / savings accounts etc).

**Comment by rrpawar96** [comment]
@ptuomola, Comment has been addressed, Please re-review the PR.

**Review by ptuomola — APPROVED** [review]
LGTM... let's wait and see if all the tests pass

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java (hunk 1 above)
Comment (rrpawar96) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: Ya, In the latest commit, the changes in those SQL scripts have been reversed, and no new changes made.

**Pull request merged**